### DISCUSSION TITLE
Puppy-raffle [H-4] selectWinner issue

### DISCUSSION THREAD

**Discussion by typicalHuman** [Cyfrin/security-and-auditing-full-course-s23#52] [Q&A]
Hello everyone, I'm currently in section 4, and I don't quite understand why in the audit it is mentioned that the attacker can "forever" halt the selectWinner function. In theory, to achieve this, all players must be contracts owned by the attacker; otherwise, there is still a chance that when selectWinner is called, the random selection might pick a regular account, and the round will end. So, it seems that this error doesn't block the code permanently but rather complicates the process of selecting a winner, is that correct?

**Comment by PatrickAlphaC** [comment]
Well, there were a few ways this could happen. Which finding in particular are you referring to?

**Reply by typicalHuman** [reply]
I mean bruteforce the selectWinner method until there is a winner without malicious code and then the raffle is over. I was just curious to know if there is really no way out (as stated in the audit) if a contract with malicious code is the winner or not?
